web/src/typeahead_helper.ts

```diff
@@ -1072,6 +1073,30 @@ export function query_matches_person(
             return true;
         }
 
+        if (allow_custom_profile_field_matching) {
+            // Check custom profile fields that are enabled for use_for_user_matching
+            const field_types = realm.custom_profile_field_types;
+            for (const field of realm.custom_profile_fields) {
+                if (
+                    field.use_for_user_matching &&
+                    (field.type === field_types.SHORT_TEXT.id ||
+                        field.type === field_types.EXTERNAL_ACCOUNT.id)
```

Comment:
I don't think we need to check `field.type` here as `use_for_user_matching` should be true only for these types.